src/commands/fal.rs

```diff
@@ -0,0 +1,215 @@
+    FormattedText, InlineKeyboardButton, InlineKeyboardButtonTypeUrl, InputFileLocal,
+    InputMessagePhoto, ReplyMarkupInlineKeyboard,
+};
+use tempfile::NamedTempFile;
+
+use crate::commands::{CommandError, CommandResult, CommandTrait};
+use crate::utilities::api_utils::DetectServerError;
+use crate::utilities::command_context::CommandContext;
+use crate::utilities::convert_argument::{ConvertArgument, StringGreedyOrReply};
+use crate::utilities::message_entities::{formatted_text, ToEntity};
+use crate::utilities::rate_limit::RateLimiter;
+use crate::utilities::text_utils;
+
+pub struct Fal {
+    command_names: &'static [&'static str],
+    description: &'static str,
+    model_name: &'static str,
+    size: (u16, u16),
+    num_inference_steps: u8,
+}
+
+impl Fal {
+    pub const fn realistic_vision() -> Self {
+        Self {
+            command_names: &["img", "rv", "image"], // this is the main model
+            description: "generate images using Realistic Vision",
+            model_name: "SG161222/RealVisXL_V4.0",
+            size: (1024, 1024),
+            num_inference_steps: 6,
+        }
+    }
+
+    pub const fn sdxl_lightning() -> Self {
+        Self {
+            command_names: &["sdxl"],
+            description: "generate images using SDXL Lightning",
+            model_name: "fast-lightning-sdxl",
+            size: (1024, 1024),
+            num_inference_steps: 4,
+        }
+    }
+}
+
+#[async_trait]
+impl CommandTrait for Fal {
+    fn command_names(&self) -> &[&str] {
+        self.command_names
+    }
+
+    fn description(&self) -> Option<&'static str> {
+        Some(self.description)
+    }
+
+    fn rate_limit(&self) -> RateLimiter<i64> {
+        RateLimiter::new(3, 60)
+    }
+
+    async fn execute(&self, ctx: &CommandContext, arguments: String) -> CommandResult {
+        let StringGreedyOrReply(prompt) = ConvertArgument::convert(ctx, &arguments).await?.0;
+
+        if let Some(issue) = text_utils::check_prompt(&prompt) {
+            log::info!("prompt rejected: {issue:?}");
+            Err(issue)?;
+        }
+
+        ctx.send_typing().await?;
+
+        let generation = self.generate(ctx, prompt).await.map_err(|e| {
+            log::error!("fal.ai error: {:#?}", e);
+            e
+        })?;
+        let image = process_image(&generation);
+        let mut temp_file = NamedTempFile::new().unwrap();
+        image.write_to(&mut BufWriter::new(&mut temp_file), ImageFormat::Jpeg).unwrap();
+
+        let message = ctx
+            .reply_custom(
+                InputMessageContent::InputMessagePhoto(InputMessagePhoto {
+                    photo: InputFile::Local(InputFileLocal {
+                        path: temp_file.path().to_str().unwrap().into(),
+                    }),
+                    thumbnail: None,
+                    added_sticker_file_ids: Vec::new(),
+                    width: image.width().try_into().unwrap(),
+                    height: image.height().try_into().unwrap(),
```

Comment:
you can pass `0` and telegram will figure the values out during server-side image processing